internal/testhelpers/session.go

```diff
@@ -136,92 +178,261 @@ func NewHTTPClientWithSessionToken(t *testing.T, reg *driver.RegistryDefault, se
+//		client := NewHTTPClientBuilder(t).
+//			SetReqestFromWhoAmI().
+//			SetIdentityFromNew().
+//			SetSessionDefault(UnrivilegedProvider()).
+//			RecordSessionMutator(reg, &sessionMutator).
+//			ClientWithSessionToken()
+//
+//		// Run some tests using `client` requiring the session to be unprivileged
+//		...
+//
+//		// Pretend the session has been refreshed, and is now privileged for an additional minute
+//		sessionMutator(func(session *session.Session) {
+//			session.SetPrivilegedUntil(time.Now().Add(time.Minute))
+//		})
+//
+//		// Run some tests using `client` requiring the session to be privileged
+//		...
+//	}
+func (c *HTTPClientBuilder) RecordSessionMutator(reg *driver.RegistryDefault, funcRef *func(func(session *session.Session))) *HTTPClientBuilder {
+	*funcRef = func(mutateFunc func(session *session.Session)) {
+		mutateFunc(c.session)
+		maybePersistSession(c.t, reg, c.session)
 	}
-	assert.Len(t, gjson.GetBytes(sess, "authentication_methods").Array(), 1+len(methods))
+	return c
 }
 
-func NewAuthorizedTransport(t *testing.T, reg *driver.RegistryDefault, sess *session.Session) *x.TransportWithHeader {
-	maybePersistSession(t, reg, sess)
+// Builder Consumers
 
-	return x.NewTransportWithHeader(http.Header{
-		"Authorization": {"Bearer " + sess.Token},
+// A generic function for creating a new client using the builders session.
+func (c *HTTPClientBuilder) ClientCallback(clientFunc func(*testing.T, *session.Session) *http.Client) *http.Client {
+	return clientFunc(c.t, c.session)
+}
+
+// A quality of life function that consumes the builder, creating a client with a session token
+// via the `NewHTTPClientWithSessionToken` function.
+func (c *HTTPClientBuilder) ClientWithSessionToken(reg *driver.RegistryDefault) *http.Client {
+	c.nilSessionCheck()
+	return c.ClientCallback(func(t *testing.T, s *session.Session) *http.Client {
+		return NewHTTPClientWithSessionToken(t, reg, s)
+	})
+}
+
+// A quality of life function that consumes the builder, creating a client with a session cookie
+// via the `NewHTTPClientWithSessionCookie` function.
+func (c *HTTPClientBuilder) ClientWithSessionCookie(reg *driver.RegistryDefault) *http.Client {
+	c.nilSessionCheck()
+	return c.ClientCallback(func(t *testing.T, s *session.Session) *http.Client {
+		return NewHTTPClientWithSessionCookie(t, reg, s)
+	})
+}
+
+// A quality of life function that consumes the builder, creating a client with a session cookie
+// without redirections via the `NewNoRedirectHTTPClientWithSessionCookie` function.
+func (c *HTTPClientBuilder) ClientNoRedirectWithSessionCookie(reg *driver.RegistryDefault) *http.Client {
+	c.nilSessionCheck()
+	return c.ClientCallback(func(t *testing.T, s *session.Session) *http.Client {
+		return NewNoRedirectHTTPClientWithSessionCookie(t, reg, s)
 	})
 }
+
+// Common configurations
+
+func NewIdentityClientWithSessionToken(t *testing.T, reg *driver.RegistryDefault, id *identity.Identity) *http.Client {
```

Comment:
These functions provide essentially the same interface as the the `NewHTTPClientWithIdentitySessionCookie` functions that were removed. The change in name is supposed to help communicate how the client is being built.